share/doc/wake/logging.md

```diff
@@ -0,0 +1,111 @@
+|--verbose          |          |    x    |    x    |     x     |      x     |    x     |          |                 |
+|--quiet            |          |         |         |           |            |    x     |          |                 |
+|--no-tty<sup>*</sup>|         |         |    x    |     x     |      x     |    x     |          |                 |
+|(default)          |          |         |         |     x     |      x     |    x     |          |                 |
+
+<sup>*</sup> `--no-tty` also prevents any coloring from being applied.
+
+
+Unless overridden with `--stderr=...` the *only* thing that ever appears on stderr coming out of wake is `logError`,
+and it always appears there.
+
+The additional command line arguments `--fd:3`, `--fd:4`, `--fd:5` can be used to output to file descriptors 3, 4, 5.
+One example shell command would be, for some wake code that had `mkLogLevel "foo"`:
+
+```
+wake --fd3="foo" mywakecode 3&>foo.txt
+```
+
+These can be concatenated with commas and include the normal log levels, so to capture both `mkLogLevel "foo"` and `logInfo` to a file:
+
+```
+wake --fd3="foo,info" 3&>foo.txt
+```
+
+## What Gets Sent To Loggers
+
+### Job Execution
+
+Every `Job` run by wake tracks its own stdout and stderr,
+and when the underlying process prints to stdout or stderr,
+it will go to the `Job`-specific stdout and stderr,
+not the stdout and stderr of the `wake` process itself.
+Every `Job` also has an "echo", which is the command that the job is executing.
+
+When writing your wake code,
+you can control which loggers these `Job`-specific stdout, stderr, and echo are sent to.
+This will influence whether they appear on the `wake` process's own stdout,
+stderr, or other arbitrary file descriptors.
+These can be controlled with `setPlanStdout`, `setPlanStderr`, and `setPlanEcho` before you `runJob`.
+These default to `logInfo`, `logError`, and `logEcho`, respectively.
+
+Regardless of to which logger you direct the `Job`'s `Stderr`, `Stdout`, and `Echo`,
+`wake` will store all the text in its database,
+discarding the information about the logger it was destined for at the time it was stored.
+It will however know whether it came from the `Job`'s stderr/stdout/echo.
+
+
+### Other Things that Appear on Wake's Output Streams During Execution
+
+We have already covered the stdout, stderr, and echo of executed `Job`s.
+
+The results of calling `print`/`println` within `wake` code are sent to `logReport` by default.
+To send them somewhere else, use `printLevel`, `printlnLevel` instead.
+
+The final output of whatever wake function or expression is called is always directed to
+wake's `stdout` unless `--quiet` is set.
+This is outside of the logger mechanism, so it is not possible to redirect this within wake
+code or with command line flags.
+
+Similarly, the "spinner" and progress bar is always sent to wake's `stdout` unless `--no-tty`
```

Comment:
Are the "spinner" and "progress bar" two different things? I can only think of one animated thing I see, which I guess must be the progress bar, but sometimes people use "spinner" as a generic term that can also mean a "progress bar" (that might be my web development influence).

Reply:
I'm not 100% sure if this sentence is true about going to stdout...

Reply:
(I made it just say progress bar, still not sure about the --quiet)